src/app/core/data/metadata-field-data.service.ts

```diff
@@ -0,0 +1,118 @@
+export class MetadataFieldDataService extends DataService<MetadataField> {
+  protected linkPath = 'metadatafields';
+  protected searchBySchemaLinkPath = 'bySchema';
+
+  constructor(
+    protected requestService: RequestService,
+    protected rdbService: RemoteDataBuildService,
+    protected store: Store<CoreState>,
+    protected halService: HALEndpointService,
+    protected objectCache: ObjectCacheService,
+    protected comparator: DefaultChangeAnalyzer<MetadataField>,
+    protected http: HttpClient,
+    protected notificationsService: NotificationsService) {
+    super();
+  }
+
+  /**
+   * Find metadata fields belonging to a metadata schema
+   * @param schema        The metadata schema to list fields for
+   * @param options       The options info used to retrieve the fields
+   * @param linksToFollow List of {@link FollowLinkConfig} that indicate which {@link HALLink}s should be automatically resolved
+   */
+  findBySchema(schema: MetadataSchema, options: FindListOptions = {}, ...linksToFollow: Array<FollowLinkConfig<MetadataField>>) {
+    const optionsWithSchema = Object.assign(new FindListOptions(), options, {
+      searchParams: [new RequestParam('schema', schema.prefix)]
+    });
+    return this.searchBy(this.searchBySchemaLinkPath, optionsWithSchema, ...linksToFollow);
+  }
+
+  /**
+   * Create or Update a MetadataField
+   *  If the MetadataField contains an id, it is assumed the field already exists and is updated instead
+   *  Since creating or updating is nearly identical, the only real difference is the request (and slight difference in endpoint):
+   *  - On creation, a CreateRequest is used
+   *  - On update, a PutRequest is used
+   * @param field    The MetadataField to create or update
+   */
+  createOrUpdateMetadataField(field: MetadataField): Observable<RemoteData<MetadataField>> {
+    const isUpdate = hasValue(field.id);
+
+    if (isUpdate) {
+      return this.put(field);
+    } else {
+      return this.create(field, new RequestParam('schemaId', field.schema.id));
+    }
+  }
+
+  /**
+   * Clear all metadata field requests
+   * Used for refreshing lists after adding/updating/removing a metadata field from a metadata schema
+   */
+  clearRequests(): Observable<string> {
+    return this.getBrowseEndpoint().pipe(
+      tap((href: string) => {
+        this.requestService.removeByHrefSubstring(href);
+      })
+    );
+  }
+
+  /**
+   * Make a new FindListRequest with given search method
+   *
+   * @param searchMethod The search method for the object
+   * @param options The [[FindListOptions]] object
+   * @param linksToFollow The array of [[FollowLinkConfig]]
+   * @return {Observable<RemoteData<PaginatedList<MetadataField>>}
+   *    Return an observable that emits response from the server
+   */
+  searchBy(searchMethod: string, options: FindListOptions = {}, ...linksToFollow: Array<FollowLinkConfig<MetadataField>>): Observable<RemoteData<PaginatedList<MetadataField>>> {
```

Comment:
Maybe overwriting the entire method may not be the right approach. We could change the `searchBy` method of the data service in the way it works like for other [find methods](https://github.com/DSpace/dspace-angular/blob/ee9f5ec7f1d9b5af20f36460f8723cfe6961de50/src/app/core/data/data.service.ts#L243). In case of `responseMsToLive` has no value it should use the current value `10 * 1000`

Reply:
Done